clis/teliod/src/qnap.rs

```diff
@@ -0,0 +1,239 @@
+use std::{
+    env, fs,
+    io::{self, Read},
+    process::{Command, Stdio},
+};
+
+use serde_json::json;
+
+use crate::config::TeliodDaemonConfig;
+
+struct PathConfig;
+
+impl PathConfig {
+    const APP_TMP_DIR: &'static str = "/tmp/nordsecuritymeshnet";
+    const QPKG_DIR: &'static str = "/share/CACHEDEV1_DATA/.qpkg/NordSecurityMeshnet";
+
+    pub fn teliod_bin() -> String {
+        format!("{}/teliod", Self::QPKG_DIR)
+    }
+
+    pub fn teliod_cfg() -> String {
+        format!("{}/teliod.cfg", Self::QPKG_DIR)
+    }
+
+    pub fn teliod_log() -> String {
+        "/var/log/teliod.log".to_owned()
+    }
+
+    pub fn meshnet_log() -> String {
+        format!("{}/meshnet.log", Self::QPKG_DIR)
+    }
+
+    pub fn pid() -> String {
+        format!("{}/teliod.pid", Self::APP_TMP_DIR)
+    }
+}
+
+#[derive(Debug, PartialEq)]
+enum HttpMethod {
+    Get,
+    Post,
+    Patch,
+    Unsupported,
+}
+
+impl HttpMethod {
+    fn from_method(method: &str) -> Self {
+        match method {
+            "GET" => HttpMethod::Get,
+            "POST" => HttpMethod::Post,
+            "PATCH" => HttpMethod::Patch,
+            _ => HttpMethod::Unsupported,
+        }
+    }
+
+    fn get_post_data(&self) -> String {
+        let mut data = String::new();
+        if matches!(self, HttpMethod::Post | HttpMethod::Patch) {
+            let content_length = env::var("CONTENT_LENGTH")
+                .ok()
+                .and_then(|len| len.parse::<usize>().ok());
+            if let Some(length) = content_length {
+                let _ = io::stdin().take(length as u64).read_to_string(&mut data);
+            }
+        }
+        data
+    }
```

Comment:
This looks like a bad design, the struct doesn't really own these data, it just loads it from the variable, so I see no reason to make it a `get` method on self

Reply:
> the struct doesn't really own these data
The method behaviour depends on the enum type, so it made sense to make it an enum method. If I turn it into a function I'd still check it against Post/Patch